### DISCUSSION TITLE
Rotate not actually 3D

### DISCUSSION THREAD

**Discussion by michaelmannucci** [sambauers/tailwindcss-3d#165] [Q&A] [answered]
How do I do an actual perspective rotate? When you 3d rotate something, one side should become smaller because it's further away. This plugin seems to basically just scale it on one axis, and it stays the same size on both sides throughout the whole rotation.

How it should work:
![image](https://github.com/sambauers/tailwindcss-3d/assets/55257848/5063474f-d229-4023-ac25-9ac4697bf079)

How it works:
![image](https://github.com/sambauers/tailwindcss-3d/assets/55257848/b357922c-aeee-42bd-8cf2-ec68108e0b47)

**Comment by RickCardoso** [comment]
> How do I do an actual perspective rotate? When you 3d rotate something, one side should become smaller because it's further away. This plugin seems to basically just scale it on one axis, and it stays the same size on both sides throughout the whole rotation.
> 
> How it should work: ![image](https://user-images.githubusercontent.com/55257848/278081014-5063474f-d229-4023-ac25-9ac4697bf079.png)
> 
> How it works: ![image](https://user-images.githubusercontent.com/55257848/278081157-b357922c-aeee-42bd-8cf2-ec68108e0b47.png)

That's because this plugin only adds the missing 3D transformations to the tailwind classes. What you want can be achieved with a combination of CSS transformations.

**Comment by sambauers** [comment, marked as answer]
You need to specify a perspective-* class on a parent element, for example:

```html
<div class="perspective-250">
  <img src="/some-image.jpg" class="h-40 w-80 -rotate-y-45"
</div>
```

More examples of rotate are here: https://play.tailwindcss.com/m1NO5HFfGx

See the CSS docs on `perspective` here: https://developer.mozilla.org/en-US/docs/Web/CSS/perspective
